### PR TITLE
[Add] First homework

### PR DESCRIPTION
**Pull request by Zera7**

### CODE REVIEW

File: HomeWork/Contents.swift [hunk 1]
```diff
@@ -0,0 +1,147 @@
+import Foundation 
+
+func trim(_ word:String, _ str:String)->String{
```
Comment (ZalyalovIldar) [inline HomeWork/Contents.swift]: поправь по всему коду отступы в сигнатуре метода по примеру: 
```swift 
func trim(_ word:String, _ str:String) -> String {
``` 
и название параметров везде полные плиз) не str а string или substring :)

File: HomeWork/Contents.swift [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+    var result = Array<Character>()
+
+    let wordCount = word.count
+    let strCount = str.count
+
+    if strCount == 0{
+        return word
+    }
```
Comment (ZalyalovIldar) [inline HomeWork/Contents.swift]: данную проверку можно сделать самой первой строчкой кода, чтобы дальше ничего не делать, и проверку сделать через guard: 
```swift 
guard !str.isEmpty else { return word }
```

File: HomeWork/Contents.swift [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+
+    var index = 0
+    var wordIndices = Array<String.Index>()
+
+    for item in word.indices{
+        wordIndices.append(item)
+    }
```
Comment (ZalyalovIldar) [inline HomeWork/Contents.swift]: если в for делаем какую то 1 операцию, то можно заменить на 1 строку: 
```swift 
word.indices.forEach{ wordIndices.append($0) }

File: HomeWork/Contents.swift [hunk 4]
```diff
@@ -0,0 +1,147 @@
...
+
+    while index < wordCount{
+        if wordCount - index - strCount >= 0 && str == word[wordIndices[index]...wordIndices[index + strCount - 1]]{
```
Comment (ZalyalovIldar) [inline HomeWork/Contents.swift]: длинные выражения лучше заранее выносить в константы, чтобы код был более компактным и читабельным: 
```swift 
let someCount (тут более корректное имя) = wordCount - index - strCount 
let someSubstring (тут более корректное имя) = String(word[wordIndices[index]...wordIndices[index + strCount - 1]])

if someCount >= 0 && str == someSubstring {

}
```

File: HomeWork/Contents.swift [hunk 5]
```diff
@@ -0,0 +1,147 @@
...
+            index += strCount
+        }
+        else{
```
Comment (ZalyalovIldar) [inline HomeWork/Contents.swift]: после else пробел (поправь по всему коду плиз)

File: HomeWork/Contents.swift [hunk 6]
```diff
@@ -0,0 +1,147 @@
...
+    return String(result)
+}
+
+//print(trim("aslkjfjkdfskjbkjdfsbvkjldasasd","as"))
+
+
+class Game{
+    var players = [Player]()
+
+    enum CharacterType : Int,CaseIterable {
```
Comment (ZalyalovIldar) [inline HomeWork/Contents.swift]: "CaseIterable" not bad 👍

File: HomeWork/Contents.swift [hunk 7]
```diff
@@ -0,0 +1,147 @@
...
+        case Wizard = 1
+        case Knight
+        case Assassin
+    }
+
+    func createPlayers(){
```
Comment (ZalyalovIldar) [inline HomeWork/Contents.swift]: тут тоже все методы поправь по отступам :)

File: HomeWork/Contents.swift [hunk 8]
```diff
@@ -0,0 +1,147 @@
...
+        print("choose your hero")
+        for item in CharacterType.allCases {
+            print("\(item.rawValue) - \(item)")
+        }
+        // ���� ����
+    }
+
+    func resetGameProgress(){
+        for man in players {
```
Comment (ZalyalovIldar) [inline HomeWork/Contents.swift]: сексизм нарастает))

File: HomeWork/Contents.swift [hunk 9]
```diff
@@ -0,0 +1,147 @@
...
+            man.loss = 0
+        }
+    }
+
+    class BattleField{
+        func beginBattle(men:[Unit]){
+            var livePlayers = men
+            var index = 0
+            while livePlayers.count > 1{
+                let attackTo = /* ��� ������ ��� ���� ���� */0
```
Comment (ZalyalovIldar) [inline HomeWork/Contents.swift]: лучше убрать коммент

File: HomeWork/Contents.swift [hunk 10]
```diff
@@ -0,0 +1,147 @@
...
+            man.wins = 0
+            man.loss = 0
+        }
+    }
+
+    class BattleField{
+        func beginBattle(men:[Unit]){
+            var livePlayers = men
+            var index = 0
+            while livePlayers.count > 1{
```
Comment (ZalyalovIldar) [inline HomeWork/Contents.swift]: отступ после "1"

File: HomeWork/Contents.swift [hunk 11]
```diff
@@ -0,0 +1,147 @@
...
+
+    func resetGameProgress(){
+        for man in players {
+            man.wins = 0
+            man.loss = 0
+        }
+    }
+
+    class BattleField{
+        func beginBattle(men:[Unit]){
```
Comment (ZalyalovIldar) [inline HomeWork/Contents.swift]: отступ сверху перед функцией и внутри функции отступы как коменты выше

File: HomeWork/Contents.swift [hunk 12]
```diff
@@ -0,0 +1,147 @@
...
+            var livePlayers = men
+            var index = 0
+            while livePlayers.count > 1{
+                let attackTo = /* ��� ������ ��� ���� ���� */0
+                livePlayers[index].attack(livePlayers[attackTo])
+                if livePlayers[attackTo].health <= 0{
```
Comment (ZalyalovIldar) [inline HomeWork/Contents.swift]: как вариант можно сделать для игроков параметр "isDead" или типа того, чтобы не сверять с числом

File: HomeWork/Contents.swift [hunk 13]
```diff
@@ -0,0 +1,147 @@
...
+                index += 1
+                if index >= livePlayers.count{
+                    index = 0
+                }
+                else{
+                if let found = find(players.map({ $0.nickname }), livePlayers[0].nickname) {
+                    players[found] = livePlayers[0]
+                }
+                    toSummarize[livePlayers[0]]
+                }
```
Comment (ZalyalovIldar) [inline HomeWork/Contents.swift]: с 70-й строки до 78-ой надо наладить отступы)

File: HomeWork/Contents.swift [hunk 14]
```diff
@@ -0,0 +1,147 @@
...
+            }
+        }
+
+        func toSummarize(_ lastPlayer:Unit){
+            print("player \(lastPlayer.nickname) won!\n\n")
```
Comment (ZalyalovIldar) [inline HomeWork/Contents.swift]: отступ после сигнатуры метода

File: HomeWork/Contents.swift [hunk 15]
```diff
@@ -0,0 +1,147 @@
...
+        }
+    }
+    return String(result)
+}
+
+//print(trim("aslkjfjkdfskjbkjdfsbvkjldasasd","as"))
+
+
+class Game{
+    var players = [Player]()
```
Comment (ZalyalovIldar) [inline HomeWork/Contents.swift]: для всех классов/моделей добавь отступ сверху перед первым полем

File: HomeWork/Contents.swift [hunk 16]
```diff
@@ -0,0 +1,147 @@
...
+        }
+    }
+
+    class Unit : Player{
+        var health:Double
+        var damage:Double
+        var protection:Double
+        var agility:Double
+
+        init(_ health:Double,_ damage:Double,_ protection:Double,_ agility:Double,_ nickname:String) {
```
Comment (ZalyalovIldar) [inline HomeWork/Contents.swift]: в init лучше не делать пустыми лейблы для параметров, максимум так: 
```swift 
init(with health: Double, damage: Double, protection: Double, agility: Double, nickname: String)
```

Comment (ZalyalovIldar) [reply HomeWork/Contents.swift]: хз почему выделилось красным 🙃

File: HomeWork/Contents.swift [hunk 17]
```diff
@@ -0,0 +1,147 @@
...
+            self.health = health
+            self.damage = damage
+            self.protection = protection
+            self.agility = agility
+            super.init(nickname)
+        }
+
+        func attack(_ attackTo:Unit){
+            let damage = self.damage + self.agility * self.damage * 0.1
```
Comment (ZalyalovIldar) [inline HomeWork/Contents.swift]: в функциях/методах более чем из 1 строки, добавляем отступ сверху (тут перед 121-ой строкой)
